sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -54,7 +54,7 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
           .getOrElse(sparkSession.sessionState.conf.columnNameOfCorruptRecord)
       val jsonFiles = files.filterNot { status =>
         val name = status.getPath.getName
-        name.startsWith("_") || name.startsWith(".")
+        (name.startsWith("_") && !name.contains("=")) || name.startsWith(".")
```

Comment:
Hm.. @dongjoon-hyun Do you mind if I ask a question?

If my understanding is correct, the `name` will be `part-..` file whether the parent directory contains `_` or not. So, it would be unnecessary extra checking. It is happening in `ParquetFileFormat` as well.

Do you mind if I open a follow-up to clean up those?